frontend_server/test/server_test.dart

```diff
@@ -0,0 +1,64 @@
+import 'dart:async';
+
+import 'package:args/src/arg_results.dart';
+import 'package:frontend_server/server.dart';
+import 'package:test/test.dart';
+
+class _MockedCompiler implements CompilerInterface {
+  Future<Null> Function(String, ArgResults) compileCallback;
+  Future<Null> Function() recompileDeltaCallback;
+  void Function() acceptLastDeltaCallback;
+  void Function() rejectLastDeltaCallback;
+  void Function(Uri) invalidateCallback;
+
+  _MockedCompiler({this.compileCallback, this.recompileDeltaCallback,
```

Comment:
always put constructors first (so that it's immediately obvious if there's an implicit constructor)